shell/platform/embedder/embedder.cc

```diff
@@ -827,12 +831,18 @@ FlutterEngineResult FlutterEngineSendPlatformMessage(
     response = response_handle->message->response();
   }
 
-  auto message = fml::MakeRefCounted<flutter::PlatformMessage>(
+  fml::RefPtr<flutter::PlatformMessage> message;
+  if (flutter_message->message_size == 0) {
```

Comment:
It's not safe to make direct access into the struct with the changes you've made; you need to make locals for the message pointer and message size using SAFE_ACCESS, and then have all future uses use the locals, since an early return for truncated structs is no longer guaranteed.